### DISCUSSION TITLE
Does this mod work with a skyblock style (player created cobblegen)?

### DISCUSSION THREAD

**Discussion by Starshak3r** [null2264/CobbleGen#4] [Q&A] [answered]
I've been trying to get this mod to work on a minimal skyblock modpack I'm trying to run for personal play, but I can't seem to get my cobblegen to produce anything other than cobblestone.

I'm also pretty illiterate when it comes to configs, so I'm not sure exactly what part of the file to edit to make it work, if its even possible to work in my situation.

thanks for any help/advice on what I did wrong.

inc: my current json
[cobblegen.txt](https://github.com/null2264/CobbleGen/files/8995042/cobblegen.txt)

**Comment by null2264** [comment, marked as answer]
You accidentally put custom generator format in default generator...
```json5
"cobbleGen": [
  // the blocks stuff goes here
]
```

**Reply by Starshak3r** [reply]
ok, so how do I get the skyblock gen to puke out ores and stuff like they would do on a server? that's what I'm after :)

**Reply by null2264** [reply]
Just upload your json file to your server

**Reply by Starshak3r** [reply]
so, the default json5 file that comes packed with the mod? That's what I was using but it isn't working at all.

after about 100 blocks it's all just cobblestone

**Reply by null2264** [reply]
Make sure the server also have the mod installed

**Reply by Starshak3r** [reply]
its a solo world. not running a server currently.

**Reply by null2264** [reply]
Have you changed the format to the default generator format?
Something like this:
```json5
"cobbleGen": [
        {
          "id": "minecraft:emerald_ore",
          "weight": 5.0
        },
        {
          "id": "minecraft:diamond_ore",
          "weight": 10.0
        },
        {
          "id": "minecraft:lapis_ore",
          "weight": 10.0
        },
        {
          "id": "minecraft:gold_ore",
          "weight": 10.0
        },
        {
          "id": "minecraft:iron_ore",
          "weight": 15.0
        },
        {
          "id": "minecraft:coal_ore",
          "weight": 20.0
        },
        {
          "id": "minecraft:cobblestone",
          "weight": 30.0
        }
    ]
```

The `"cobbleGen: { "minecraft:bedrock": [ ] }` format is for custom generators with block modifiers

**Reply by Starshak3r** [reply]
ok I just pasted that into my json file. should it be .json or .json5?

**Reply by null2264** [reply]
The file should be `.json5`

**Reply by Starshak3r** [reply]
I must be totally doing something wrong then. It's  definitely not working with this skyblock mod I am running.

going to try it with everything else disabled but that one

**Reply by Starshak3r** [reply]
[20:13:12] [Render thread/INFO]: Setting user: STARSHAK3R
[20:13:12] [Render thread/ERROR]: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 2 path $
[20:13:12] [Render thread/INFO]: Encountered an error while reading cobblegen config, falling back to default values.
[20:13:12] [Render thread/INFO]: If this problem persists, delete the config file cobblegen.json5 and try again.

what does that mean?

**Reply by null2264** [reply]
That means your json file is not valid

**Reply by Starshak3r** [reply]
I found it.. ugh.. See what I meant about being json illiterate haha

**Reply by Starshak3r** [reply]
> Have you changed the format to the default generator format? Something like this:
> 
> ```js
> "cobbleGen": [
>         {
>           "id": "minecraft:emerald_ore",
>           "weight": 5.0
>         },
>         {
>           "id": "minecraft:diamond_ore",
>           "weight": 10.0
>         },
>         {
>           "id": "minecraft:lapis_ore",
>           "weight": 10.0
>         },
>         {
>           "id": "minecraft:gold_ore",
>           "weight": 10.0
>         },
>         {
>           "id": "minecraft:iron_ore",
>           "weight": 15.0
>         },
>         {
>           "id": "minecraft:coal_ore",
>           "weight": 20.0
>         },
>         {
>           "id": "minecraft:cobblestone",
>           "weight": 30.0
>         }
>     ]
> ```
> 
> The `"cobbleGen: { "minecraft:bedrock": [ ] }` format is for custom generators with block modifiers
[cobblegen.txt](https://github.com/null2264/CobbleGen/files/8996554/cobblegen.txt)
This Solved  my issue!! Thanks!! :)

**Reply by null2264** [reply]
Awesome! No problem. Thanks for using my mod...
